### PR TITLE
Add livereload support

### PR DESCRIPTION
**Pull request by dwt**
As promised. :)

Some things I'm missing: How does the js combining and sass compilation work? Is there a dependency missing?

I wasn't able to get the http server in app.js to bind to 0.0.0.0 - are there special tricks involved?

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -78,7 +78,6 @@ var tpl = function(file, data, res) {
 			}
 
 		});
-
```
Comment (xi) [inline app.js:81]: Please do not change unrelated formatting.

File: watch_requirements.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+formic # only python2 compatibile
```
Comment (xi) [inline watch_requirements.txt:1]: According to [pep8](https://www.python.org/dev/peps/pep-0008/#inline-comments), two spaces should be separated by at least two spaces.

File: bin/initialize_db.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
+
+cd "$(dirname "$0")/.."
+mkdir -p data/postgres
+if test ! -d  data/postgres/base; then
+	pg_ctl initdb -D data/postgres
+	pg_ctl start -w -D data/postgres -o "-h localhost"
+	createdb voterunner
+	pg_ctl stop -D data/postgres
+fi
```
Comment (xi) [inline bin/initialize_db.sh:10]: Would it make sense to also set up the table in this file?

File: README.md [hunk 4]
```diff
@@ -69,14 +69,14 @@ Voterunner is a [node.js](http://nodejs.org/) app using
...
     $ cd voterunner
     $ npm install
-    $ export PORT=5000
-    $ export DATABASE_URL='postgresql://user:password@host/database'
-    $ node app.js
-        info  - socket.io started
-        info  - Listening on 5000
+    $ pip install -r watch_requirements.txt
+    $ bin/initialize_db.sh
+    $ bin/watch.py
```
Comment (xi) [inline README.md:77]: IMHO watching should be optional. Could you please restore something like the original text (now including `initialize_db.sh`) and add an extra section about live reload?

File: README.md [hunk 5]
```diff
@@ -69,14 +69,14 @@ Voterunner is a [node.js](http://nodejs.org/) app using
...
+    $ open http://localhost:5000/ # introduction
+    $ open http://localhost:5000/my-topic/ # discuss on a topic
```
Comment (xi) [inline README.md:79]: On my system, `open` is an alias for `openvt`. The equivalent to what you are referring to is probably `xdg-open`. Could you try to make this more platform-independent?

**Comment by xi** [comment]
> How does the js combining and sass compilation work? Is there a dependency missing?

I added the relevant documentation in 8027910. (Note: the requirements are not included in `package.json` to avoid duplicating them in every project; node-sass especially is a large binary)

> I wasn't able to get the http server in app.js to bind to 0.0.0.0

c5cfd7a should do the trick.

File: bin/watch.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python
```
Comment (xi) [inline bin/watch.py:1]: Should be `#!/usr/bin/env python2`, right?

**Comment by xi** [comment]
The `initialize_db.sh` script did not work for me. The [`pg_ctl` command was not in the path](https://askubuntu.com/questions/385416/pg-ctl-command-not-found-what-package-has-this-command) (debian jessie). When I added `/usr/lib/postgresql/9.4/bin/` to the PATH I got the following error:

```
waiting for server to start....LOG:  could not bind IPv6 socket: Address already in use
HINT:  Is another postmaster already running on port 5432? If not, wait a few seconds and retry.
LOG:  could not bind IPv4 socket: Address already in use
HINT:  Is another postmaster already running on port 5432? If not, wait a few seconds and retry.
WARNING:  could not create listen socket for "localhost"
FATAL:  could not create any TCP/IP sockets
 stopped waiting
pg_ctl: could not start server
Examine the log output.
createdb: could not connect to database template1: FATAL:  role "tobias" does not exist
pg_ctl: PID file "data/postgres/postmaster.pid" does not exist
Is server running?
```

... I really like the idea of a local database though.

**Comment by xi** [comment]
The live reload did not work for me (tested on chrome). I could see no reaction when changing a file (neither in the console nor in the browser).

File: bin/initialize_db.sh (hunk 3 above)
Comment (dwt) [reply bin/initialize_db.sh:10]: I would like that - but that has to be coded first. Right now the app does it on demand, which is fine with me.

File: README.md (hunk 4 above)
Comment (dwt) [reply README.md:77]: will do.

File: README.md (hunk 5 above)
Comment (dwt) [reply README.md:79]: Sorry, there is no platform independent way to do this, that I'm aware of. Probably

```
$ # open http://localhost:5000/ # for the introduction
```

is the best that can be done here.

**Comment by dwt** [comment]
regarding live reload not doing anything for you in the browser: Did you install and activate the browser plugin from http://livereload.com/extensions/ ?

**Comment by dwt** [comment]
Regarding local database: I'm not sure how to do this (portable) on debian. If the binaries are not in the path and you already have the db running as a service, it is probably easiest for you to continue to do so. I'm learning towards adding a flag to the relevant scripts wether they should try to start the db locally.
